cmd/gitannex/gitannex_test.go

```diff
@@ -313,14 +320,109 @@ func getUniqueRemoteName(t *testing.T) string {
 	return fmt.Sprintf("remote-%x", sha256.Sum256([]byte(t.Name())))
 }
 
+// Drop-in replacement for `filepath.Rel()` that works around a Windows-specific
+// quirk when one of the paths begins with `\\?\` or `//?/`. It seems that
+// fstest gives us paths with this prefix on Windows, which throws a wrench in
+// the gitannex tests that need to construct relative paths from absolute paths.
+// For a demonstration, see `TestWindowsFilepathRelQuirk` below.
+//
+// The `\\?\` prefix tells Windows APIs to pass strings unmodified to the
+// filesystem without additional parsing [1]. Our workaround is roughly to add
+// the prefix to whichever parameter doesn't have it (when the OS is Windows).
+// I'm not sure this generalizes, but it works for the the kinds of inputs we're
+// throwing at it.
+//
+// [1]: https://learn.microsoft.com/en-us/windows/win32/fileio/naming-a-file?redirectedfrom=MSDN#win32-file-namespaces
+func relativeFilepathWorkaround(basepath, targpath string) (string, error) {
+	if runtime.GOOS != "windows" {
+		return filepath.Rel(basepath, targpath)
+	}
+	// Canonicalize paths to use backslashes.
+	basepath = filepath.Clean(basepath)
+	targpath = filepath.Clean(targpath)
+
+	const winFilePrefixDisableStringParsing = `\\?\`
+	baseHasPrefix := strings.HasPrefix(basepath, winFilePrefixDisableStringParsing)
+	targHasPrefix := strings.HasPrefix(targpath, winFilePrefixDisableStringParsing)
+
+	if baseHasPrefix && !targHasPrefix {
+		targpath = winFilePrefixDisableStringParsing + targpath
+	}
+	if !baseHasPrefix && targHasPrefix {
+		basepath = winFilePrefixDisableStringParsing + basepath
+	}
+	return filepath.Rel(basepath, targpath)
```

Comment:
I think that `\\?\` only makes sense at the front of *absolute* file paths. This function returns a relative path, which is why I omitted it. But I admit I'm totally out of my depth with Windows file paths and could be confused.

> Perhaps this is actually a bug in filepath.Rel()?

This crossed my mind :) I may try filing an upstream bug on Monday!

Reply:
I'm hoping with the `fspath.SplitFs` suggestion below we can get rid of this code entirely. There are quite a few oddities with Windows paths which the local backend understands and it would be nice not to duplicate here!

Reply:
Sadly, I don't know how to get rid of it because some of the tests need to construct a relative path from the fstest-provided absolute paths. I made sure to write a test that replicates the weird behavior. If that test ever starts to fail (i.e. it is a bug and has been fixed upstream), we can delete the workaround function and use `filepath.Rel()` directly in the tests.

Reply:
I guess the obvious question here is why do the tests need a relative path?

Reply:
Ah, test coverage for relative paths is necessary because git-annex typically (always?) passes in relative paths. For instance, it will give us the command `TRANSFER STORE SomeKey .git/annex/objects/$RestOfPath`.

If you'd prefer to avoid the weird workaround function, I suppose I could just skip the affected tests when the OS is Windows. WDYT?

Reply:
I see. The whole windows path thing is a can of worms! Lets keep your workaround and maybe some flash of inspiration will hit us in the future :-)

Reply:
It sure is! FYI, I filed an upstream bug: <https://github.com/golang/go/issues/68847>.